README.md

```diff
@@ -1,61 +1,419 @@
 # Abstract
 
-Basic definitions for abstract algebraic structures in Swift.
+A take on abstract algebraic structures, in Swift.
 
-To each structure are assigned some laws, that can be verified with property-based testing (and are verified for all the types considered in this repo using SwiftCheck), so than anyone can give some structure to more types: the only thing that's required is to verify the laws.
+------
+
+`Abstract` is a Swift library that defines protocols for the main [abstract algebraic structures](https://en.wikipedia.org/wiki/Abstract_algebra), along with some concrete implementations for them that declare specific semantics.
```

Comment:
Nit: `s/for the main/for common`

>along with some concrete implementations for them that declare specific semantics.

along with some concrete implementations for Swift datatypes.